src/common.cpp

```diff
@@ -1993,3 +1994,27 @@ void redirect_tty_output() {
     show_stackframe(L'E', 99, 1);
     abort();
 }
+
+/// Return a path to a directory where we can store temporary files.
+wcstring get_path_to_tmp_dir()
+{
+    env_var_t env_tmpdir = env_get_string(L"TMPDIR");
+    if (!env_tmpdir.missing()) return env_tmpdir;
+
+#if defined(_CS_DARWIN_USER_TEMP_DIR)
+    char osx_tmpdir[PATH_MAX];
+    if(confstr(_CS_DARWIN_USER_TEMP_DIR, osx_tmpdir, PATH_MAX - 1))
+    {
+        return str2wcstring(osx_tmpdir);
+    }
```

Comment:
If `confstr` returns `0` you'll end up hitting the end of the function without a `return` statement. You should be seeing a warning about this from the compiler.